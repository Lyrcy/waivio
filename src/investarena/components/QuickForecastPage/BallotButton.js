import React from 'react';
import PropTypes from 'prop-types';

import BallotTimer from './BallotTimer';

import './BallotButton.less';

const BallotButton = props => (
    <div className="ballotButton__container">
      <div className="ballotButton__button-container">
        <button disabled={props.disabled} onClick={() => props.onClickCB('yes', props.permlink)} className='ballotButton ballotButton__positive'>
          {props.positiveText}
        </button>
        <button disabled={props.disabled} onClick={() => props.onClickCB('no',props.permlink)} className='ballotButton ballotButton__negative'>
          {props.negativeText}
        </button>

      </div>
      <div className="ballotButton__timer">
        <BallotTimer/>
      </div>
    </div>
  );

BallotButton.propTypes = {
  onClickCB: PropTypes.func,
  positiveText: PropTypes.string,
  negativeText: PropTypes.string,
  permlink: PropTypes.string,
  disabled: PropTypes.bool.isRequired,
};

BallotButton.defaultProps = {
  onClickCB: () => {
  },
  positiveText: 'Yes',
  negativeText: 'No',
  permlink: '',
};

export default BallotButton;
