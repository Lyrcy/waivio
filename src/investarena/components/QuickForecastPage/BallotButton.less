@import (reference) '../../../client/styles/custom.less';

.ballotButton {
  padding: 0 10px;
  border-radius: 4px;
  cursor: pointer;
  font-family: @font-family-sans-serif;
  background-color: transparent;
  transition: .5s;
  width: 100px;
  border: none;

  &:disabled {
    cursor: auto;
  }

  &__container {
    display: flex;
    justify-content: space-between;
    align-items: baseline;
  }

  &__button-container {
    display: flex;
    justify-content: center;
    align-items: flex-end;
    height: 50%;
    flex-grow: 3;
    padding-right: 20px;
  }

  &__positive {
    border: 1px solid @green-darker;
    color: @green-darker;
    margin: 0 10px 0 0;

    &:hover {
      background-color: @green-darker;
      color: @white;
    }

    &:disabled {
      &:hover {
        background-color: transparent;
        color: @green-darker;
      }
    }
  }

  &__negative {
    border: 1px solid @red;
    color: @red;
    margin: 0 0 0 10px;

    &:hover {
      background-color: @red;
      color: @white;
    }

    &:disabled {
      &:hover {
        background-color: transparent;
        color: @red;
      }
    }
  }

  &__timer {
    font-family: @font-family-monospace;
  }

  &::first-letter {
    text-transform: uppercase;
  }
}
