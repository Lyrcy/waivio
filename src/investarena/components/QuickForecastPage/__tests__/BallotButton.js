import React from 'react';
import { shallow } from 'enzyme';

import BallotButton from '../BallotButton';

describe('<BallotButton />', () => {
  const props = {
    onClickCB: jest.fn(),
    positiveText: 'Yes',
    negativeText: 'No',
  };
  const wrapper = shallow(<BallotButton {...props} />);

  it('renders without exploding', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('when i click in button "yes" onClicl CB has been called', () => {
    wrapper.find('.ballotButton__positive').simulate('click');
    expect(props.onClickCB).toHaveBeenCalled()
  });

  it('when i click in button "no" onClicl CB has been called', () => {
    wrapper.find('.toHaveBeenCalled').simulate('click');
    expect(props.onClickCB).toHaveBeenCalled()
  });
});


