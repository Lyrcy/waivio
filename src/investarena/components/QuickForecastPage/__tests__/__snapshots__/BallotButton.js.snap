// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallotButton /> renders without exploding 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BallotButton
    negativeText="No"
    onClickCB={[MockFunction]}
    positiveText="Yes"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ballotButton__button-container"
        >
          <button
            className="ballotButton ballotButton__positive"
            onClick={[MockFunction]}
          >
            Yes
          </button>
          <button
            className="ballotButton toHaveBeenCalled"
            onClick={[MockFunction]}
          >
            No
          </button>
        </div>,
        <div
          className="ballotButton__timer"
        >
          <BallotTimer />
        </div>,
      ],
      "className": "ballotButton__container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="ballotButton ballotButton__positive"
              onClick={[MockFunction]}
            >
              Yes
            </button>,
            <button
              className="ballotButton toHaveBeenCalled"
              onClick={[MockFunction]}
            >
              No
            </button>,
          ],
          "className": "ballotButton__button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Yes",
              "className": "ballotButton ballotButton__positive",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Yes",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No",
              "className": "ballotButton toHaveBeenCalled",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "No",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <BallotTimer />,
          "className": "ballotButton__timer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ballotButton__button-container"
          >
            <button
              className="ballotButton ballotButton__positive"
              onClick={[MockFunction]}
            >
              Yes
            </button>
            <button
              className="ballotButton toHaveBeenCalled"
              onClick={[MockFunction]}
            >
              No
            </button>
          </div>,
          <div
            className="ballotButton__timer"
          >
            <BallotTimer />
          </div>,
        ],
        "className": "ballotButton__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="ballotButton ballotButton__positive"
                onClick={[MockFunction]}
              >
                Yes
              </button>,
              <button
                className="ballotButton toHaveBeenCalled"
                onClick={[MockFunction]}
              >
                No
              </button>,
            ],
            "className": "ballotButton__button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Yes",
                "className": "ballotButton ballotButton__positive",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "Yes",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No",
                "className": "ballotButton toHaveBeenCalled",
                "onClick": [MockFunction],
              },
              "ref": null,
              "rendered": "No",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <BallotTimer />,
            "className": "ballotButton__timer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
